### PR TITLE
Features/frontend app directory

### PR DESCRIPTION
**Pull request by maximihajlov**
This pull request migrates all the Next.js pages from the old `pages/` directory router to the new `app/` directory router. 

The main feature of `app/` directory router is the new React Server Components. Old server-side methods were replaced with these RSC, which made code much cleaner and easier to maintain. 

I've also enabled Edge runtime, which is more suitable for our case and make it much easier to deploy to the serverless infrastructure. All core functionality works locally and in docker, so I think it's stable enough to merge.

Feel free to write comments and report bugs! Feedback is highly appreciated.

### CODE REVIEW

**Comment by healingdrawing** [comment]
are you sure this is safe? `"^13.4.3"`
 i mean all versions starts from 13.
it is not a complainment.
 but is there chance that they make any changes incompatible with previous version , but still under the 13?

**Comment by healingdrawing** [comment]
how this works?
`textarea.value = value.substring(0, start) + "\t" + value.substring(end)`

**Comment by maximihajlov** [comment]
> are you sure this is safe? `"^13.4.3"` i mean all versions starts from 13. it is not a complainment. but is there chance that they make any changes incompatible with previous version , but still under the 13?

Hey, thanks for the feedback. 

Yes, it's totally safe. This syntax allows only minor releases which are always backward compatible. Backward incompatible releases will be marked as major releases and are not allowed with this dependency declaration.

https://docs.npmjs.com/about-semantic-versioning

**Comment by maximihajlov** [comment]
> how this works? `textarea.value = value.substring(0, start) + "\t" + value.substring(end)`

Where is it?

File: frontend/src/api/posts/edge.ts
Comment (bomanviktor) [inline frontend/src/api/posts/edge.ts:1]: This is really nice stuff. Can I take this on vanilla frontend?

**Review by bomanviktor — APPROVED** [review]
Very nice, learned a lot as well by reviewing this. 

Made me re-think how I should structure my FS in a better way.

Also the edge.ts was so nice. 10/10 would review again

**Review by bomanviktor — COMMENTED** [review]
Just version updates. 

Changelogs were TL;DR but they are probably cool:
https://github.com/vercel/next.js/releases

**Pull request merged**